src/actions.ts

```diff
@@ -0,0 +1,100 @@
+import InstanceSkel = require("../../../instance_skel");
+import { MagewellConfig } from "./config";
+import { CompanionActionEvent, CompanionActions } from "../../../instance_skel_types";
+import { MagewellClient } from "./client";
+import { DeviceStatus } from "./magewell";
+
+export enum ActionId {
+  Stream = 'stream',
+  Record = 'record'
+}
+
+export function HandleAction(instance: InstanceSkel<MagewellConfig>, client: MagewellClient, action: CompanionActionEvent): void {
```

Comment:
in case you're not aware, actions can do the same callback flow as feedbacks instead of needing this HandleAction function.
Its personal preference which you go with, so either is fine to do.

Reply:
I didn't know, but in this case the HandleAction function does a try/catch around the action. When I move it into the actions, I would have to duplicate that.